### PR TITLE
Always use luks_data.min_entropy as a default minimum entropy

### PR DESCRIPTION
**Pull request by poncovka**
Use `luks_data.min_entropy` if the minimum entropy is not set.

### CODE REVIEW

File: blivet/devicefactory.py [hunk 1]
```diff
@@ -332,7 +332,7 @@ def __init__(self, storage, **kwargs):
             setattr(self, setting, value)
 
         self.fstype = None  # not included in default_settings b/c of special handling below
-        self.min_luks_entropy = kwargs.get("min_luks_entropy") or luks_data.min_entropy
+        self.min_luks_entropy = kwargs.get("min_luks_entropy", luks_data.min_entropy)
```
Comment (dwlehman) [inline blivet/devicefactory.py]: The old code here is actively protecting against using a passed-in `0` or `None`. Is that not desirable?

File: blivet/formats/luks.py [hunk 2]
```diff
@@ -130,7 +130,7 @@ def __init__(self, **kwargs):
         self._key_file = kwargs.get("key_file")
         self.escrow_cert = kwargs.get("escrow_cert")
         self.add_backup_passphrase = kwargs.get("add_backup_passphrase", False)
-        self.min_luks_entropy = kwargs.get("min_luks_entropy", 0)
+        self.min_luks_entropy = kwargs.get("min_luks_entropy", luks_data.min_entropy)
```
Comment (dwlehman) [inline blivet/formats/luks.py]: If this is a real minimum maybe we should be treating as such by doing something like `min(kwargs.get('min_luks_entropy') or 0, luks_data.min_entropy)`.

Comment (dwlehman) [reply blivet/formats/luks.py]: Oops, had that backwards.

File: blivet/devicefactory.py (hunk 1 above)
Comment (vpodzime) [reply blivet/devicefactory.py]: Yeah, `None` should be replaced  by a number, I think. `0` should, however, be respected.

File: blivet/formats/luks.py (hunk 2 above)
Comment (vpodzime) [reply blivet/formats/luks.py]: It's not really an absolutely required minimum, it's more a recommended minimum.

**Comment by poncovka** [comment]
Updated.

This pull request shouldn't change the current behaviour of Blivet or Anaconda. Blivet uses by default zero minimum entropy, Anaconda uses by default `MIN_CREATE_ENTROPY`. This is not going to change. We just want to simplify the code in Anaconda (see https://github.com/rhinstaller/anaconda/pull/2061).

File: blivet/devicefactory.py [hunk 3]
```diff
@@ -332,7 +332,11 @@ def __init__(self, storage, **kwargs):
             setattr(self, setting, value)
 
         self.fstype = None  # not included in default_settings b/c of special handling below
-        self.min_luks_entropy = kwargs.get("min_luks_entropy") or luks_data.min_entropy
+        self.min_luks_entropy = kwargs.get("min_luks_entropy", None)
```
Comment (vpodzime) [inline blivet/devicefactory.py]: `None` is returned by default by `.get()` if the key doesn't exist.

File: blivet/formats/luks.py [hunk 4]
```diff
@@ -130,7 +130,10 @@ def __init__(self, **kwargs):
         self._key_file = kwargs.get("key_file")
         self.escrow_cert = kwargs.get("escrow_cert")
         self.add_backup_passphrase = kwargs.get("add_backup_passphrase", False)
-        self.min_luks_entropy = kwargs.get("min_luks_entropy", 0)
+        self.min_luks_entropy = kwargs.get("min_luks_entropy", None)
```
Comment (vpodzime) [inline blivet/formats/luks.py]: This is a change in behavior in Blivet, isn't it? It was defaulting to `0` and now it defaults to `luks_data.min_entropy`.

Comment (poncovka) [reply blivet/formats/luks.py]: But `luks_data.min_entropy` is set to `0` by default in Blivet.

File: blivet/devicefactory.py (hunk 3 above)
Comment (poncovka) [reply blivet/devicefactory.py]: Fixed.

File: blivet/formats/luks.py (hunk 4 above)
Comment (vpodzime) [reply blivet/formats/luks.py]: Oh! That explains a lot :smile:

**Comment by poncovka** [comment]
@dwlehman, is there anything else I should change?

**Pull request merged**